resolwe/flow/utils/docs/autoprocess.py

```diff
@@ -53,10 +55,24 @@ def get_process_definition_start(fname, slug):
     :rtype: int
 
     """
-    with open(fname) as file_:
-        for i, line in enumerate(file_):
-            if re.search(r"slug:\s*{}".format(slug), line):
-                return i + 1
+    if fname.endswith(".py"):
+        with open(fname) as handle:
+            for node in ast.walk(ast.parse(handle.read())):
```

Comment:
This might return bogus value for any class that has slug attribute set to the correct value.
I know this is very improbable but maybe consider checking if class we are visiting is a subclass of Process.

See https://github.com/genialis/resolwe/blob/7ac6f69bb47b76fe5a1aff421209b97d1ac15720/resolwe/process/parser.py#L165 for details.

Reply:
Maybe it would be better to traverse file only once instead of twice (second pass is just to find the starting line of the class definition). 

1. Modify ``ProcessDescriptor.Metadata`` class to include the ``lineno`` attribute (by default None).

2.  Add the line  ``descriptor.metadata.lineno = node.lineno`` in ``descriptor.py`` after line https://github.com/genialis/resolwe/blob/7ac6f69bb47b76fe5a1aff421209b97d1ac15720/resolwe/process/parser.py#L180 .

3. Use ``p.lineno`` in ``autoprocess.py`` to get the line where the class is defined.

Reply:
Thanks for suggesting this @gregorjerse , makes total sense. I've applied the suggested changes. Please re-review.